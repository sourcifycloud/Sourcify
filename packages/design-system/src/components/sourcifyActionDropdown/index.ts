import sourcifyActionDropdown from './ActionDropdown.vue';
export default sourcifyActionDropdown;
