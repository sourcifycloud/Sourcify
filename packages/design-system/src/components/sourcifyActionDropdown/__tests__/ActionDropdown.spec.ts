import { render } from '@testing-library/vue';
import sourcifyActionDropdown from '../ActionDropdown.vue';

describe('components', () => {
	describe('sourcifyActionDropdown', () => {
		it('should render default styling correctly', () => {
			const wrapper = render(sourcifyActionDropdown, {
				props: {
					items: [
						{
							id: 'item1',
							label: 'Action 1',
						},
						{
							id: 'item2',
							label: 'Action 2',
						},
					],
				},
				stubs: ['sourcify-icon', 'el-dropdown', 'el-dropdown-menu', 'el-dropdown-item'],
			});
			expect(wrapper.html()).toMatchSnapshot();
		});
		it('should render custom styling correctly', () => {
			const wrapper = render(sourcifyActionDropdown, {
				props: {
					items: [
						{
							id: 'item1',
							label: 'Action 1',
							icon: 'thumbs-up',
						},
						{
							id: 'item2',
							label: 'Action 2',
							icon: 'thumbs-down',
							disabled: true,
						},
						{
							id: 'item3',
							label: 'Action 3',
							icon: 'heart',
							divided: true,
						},
					],
				},
				stubs: ['sourcify-icon', 'el-dropdown', 'el-dropdown-menu', 'el-dropdown-item'],
			});
			expect(wrapper.html()).toMatchSnapshot();
		});
	});
});
