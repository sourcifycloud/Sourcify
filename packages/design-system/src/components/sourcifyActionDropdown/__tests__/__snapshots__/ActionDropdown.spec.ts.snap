// Vitest Snapshot v1

exports[`components > sourcifyActionDropdown > should render custom styling correctly 1`] = `
"<div class=\\"action-dropdown-container actionDropdownContainer\\">
  <el-dropdown-stub trigger=\\"click\\" size=\\"\\" hideonclick=\\"true\\" placement=\\"bottom\\" visiblearrow=\\"true\\" showtimeout=\\"250\\" hidetimeout=\\"150\\" tabindex=\\"0\\">
    <div class=\\"activator\\">
      <sourcify-icon-stub icon=\\"ellipsis-v\\" size=\\"medium\\"></sourcify-icon-stub>
    </div>
  </el-dropdown-stub>
</div>"
`;

exports[`components > sourcifyActionDropdown > should render default styling correctly 1`] = `
"<div class=\\"action-dropdown-container actionDropdownContainer\\">
  <el-dropdown-stub trigger=\\"click\\" size=\\"\\" hideonclick=\\"true\\" placement=\\"bottom\\" visiblearrow=\\"true\\" showtimeout=\\"250\\" hidetimeout=\\"150\\" tabindex=\\"0\\">
    <div class=\\"activator\\">
      <sourcify-icon-stub icon=\\"ellipsis-v\\" size=\\"medium\\"></sourcify-icon-stub>
    </div>
  </el-dropdown-stub>
</div>"
`;
